roles/lib_openshift/src/class/oadm_policy_user.py

```diff
@@ -0,0 +1,184 @@
+        ''' return whether scc exists '''
+        results = self.get()
+        if results['returncode'] == 0:
+            self.security_context_constraint = SecurityContextConstraints(results['results'][0])
+
+            if self.security_context_constraint.find_user(self.config.config_options['user']['value']) != None:
+                return True
+
+            return False
+
+        return results
+
+    def exists(self):
+        '''does the object exist?'''
+        if self.config.config_options['resource_kind']['value'] == 'cluster-role':
+            return self.exists_role_binding()
+
+        elif self.config.config_options['resource_kind']['value'] == 'role':
+            return self.exists_role_binding()
+
+        elif self.config.config_options['resource_kind']['value'] == 'scc':
+            return self.exists_scc()
+
+        return False
+
+    def perform(self):
+        '''perform action on resource'''
+        cmd = ['-n', self.config.namespace, 'policy',
+               self.config.config_options['action']['value'],
+               self.config.config_options['name']['value'],
+               self.config.config_options['user']['value']]
+
+        return self.openshift_cmd(cmd, oadm=True)
+
+    @staticmethod
+    def run_ansible(params, check_mode):
+        '''run the idempotent ansible code'''
+
+        state = params['state']
+
+        action = None
+        if state == 'present':
+            action = 'add-' + params['resource_kind'] + '-to-user'
+        else:
+            action = 'remove-' + params['resource_kind'] + '-from-user'
+
+        nconfig = OadmPolicyUserConfig(params['namespace'],
+                              params['kubeconfig'],
+                              {'action': {'value': action, 'include': False},
+                               'user': {'value': params['user'], 'include': False},
+                               'resource_kind': {'value': params['resource_kind'], 'include': False},
+                               'name': {'value': params['resource_name'], 'include': False},
+                              })
+
+        oadmpolicyuser = OadmPolicyUser(nconfig)
+        # Run the oadm policy user related command
+
+        ########
+        # Delete
+        ########
+        if state == 'absent':
+            if not oadmpolicyuser.exists():
+                return {'changed': False, 'state': 'absent'}
+
+            if check_mode:
+                return {'changed': False, 'msg': 'CHECK_MODE: would have performed a delete.'}
+
+            api_rval = oadmpolicyuser.perform()
+
+            if api_rval['returncode'] != 0:
+                return {'msg': api_rval}
+
+            return {'changed': True, 'results' : api_rval, state:'absent'}
+
+        if state == 'present':
+            ########
+            # Create
+            ########
+            results = oadmpolicyuser.exists()
+            if isinstance(results, dict) and results.has_key('returncode') and results['returncode'] != 0:
```

Comment:
To be compatible with python3.5 has_key is no longer the preferred method.

Use: `'returncode' in results`